File: challenges/2-idade-em-dias/elixir/marcosvag/idade_em_dias.exs
defmodule Converter do
    def idade_em_dias(dias_input) do
        anos = div(dias_input, 365)
        meses = rem(dias_input, 365) |> div(30)
        dias = rem(dias_input, 365) |> rem(30)
        IO.puts "#{anos} ano(s)\n#{meses} mes(es)\n#{dias} dia(s)"
    end
end 

File: challenges/2-idade-em-dias/java/Neg0023/IdadeEmDias.java
public class IdadeEmDias {
    public static void main(String[] args) {
        int idadeEmDias = Integer.parseInt(args[0]);
        System.out.printf("%d ano(s)%n",anos(idadeEmDias));
        System.out.printf("%d mes(s)%n", meses(idadeEmDias));
        System.out.printf("%d dia(s)%n", dias(idadeEmDias));
    }

    private static Integer dias(int idadeEmDias) {
        return idadeEmDias - 365 * anos(idadeEmDias) - 30 * meses(idadeEmDias);
    }

    private static Integer meses(int idadeEmDias) {
        return (idadeEmDias - 365 * anos(idadeEmDias)) / 30;
    }

    private static Integer anos(int valorEmDias) {
        return valorEmDias / 365;
    }
}
